test/check-machines-create

```diff
@@ -519,12 +519,7 @@ class TestMachinesCreate(machineslib.VirtualMachinesCase):
                                                       create_and_run=True))
 
         # Set up the PXE server configuration files
-        cmds = [
-            "mkdir -p /var/lib/libvirt/dnsmasq",
-            f"echo \"{PXE_SERVER_CFG}\" > /var/lib/libvirt/dnsmasq/pxe.cfg",
-            "chmod 666 /var/lib/libvirt/dnsmasq/pxe.cfg"
-        ]
-        self.machine.execute("; ".join(cmds))
+        self.machine.write("/var/lib/libvirt/dnsmasq/pxe.cfg", PXE_SERVER_CFG, perm="0666")
```

Comment:
This isn't already in vm_tmpdir (unlike the previous cases), so this one actually *should* use self.write_file(), so that the file is restored after the test.

Reply:
Looking deeper into this it's more of a mess then.

```
        # Set up the PXE server configuration files
        self.write_file("/var/lib/libvirt/dnsmasq/pxe.cfg", PXE_SERVER_CFG, perm="0666")

        # Define and start a NAT network with tftp server configuration
        self.machine.write("/tmp/pxe-nat.xml", NETWORK_XML_PXE)
        self.machine.execute("virsh net-define /tmp/pxe-nat.xml; virsh net-start pxe-nat")
```

Another write() although to /tmp/ we should probably clean that up. And then the network define which is never undefined. So needs cleanup to:

```
virsh net-destroy 
virsh net-undefine --network
```

Reply:
sorry -- this is /var/lib/libvirt, so this is fine.